### ISSUE TITLE
Fix Guice injection for audit logging module

### ISSUE THREAD

**Issue by sethvargo** [abcxyz/lumberjack#354]
### TL;DR

We currently use [`RemoteProcessingModule`](https://github.com/abcxyz/lumberjack/blob/5a59c9583e40ee1d305448cfa05407c6a26db3c7/clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/modules/RemoteProcessorModule.java) to inject the processor, but it relies on a GRPC client that cannot be closed. At best, this pollutes log output. At worst, we're not properly closing connections and losing data.

The Guice docs [recommend against](https://github.com/google/guice/wiki/Avoid-Injecting-Closable-Resources) storing closable resources and provide some alternative suggestions.

### Environment Details

```markdown
N/A
```
